File: Source/src_libgdx/ProjectCube/src/com/cube/main/Elements/Ball.java
package com.cube.main.Elements;

import com.cube.data.Animation;
import com.cube.main.CubeGame;

public class Ball extends PixelActor {

	public int directionX = 1;
	public int directionY = 1;
	
	@Override
	public void LoadAnimations() {
		Animation anim = CubeGame.AssetManager.get("data/ball.pmz", Animation.class);
		_currentAnimation = anim;
		_animations.add(anim);
	}

}

File: Source/src_spritekit/Project Cube/Project Cube/Ball.swift
//
//  Ball.swift
//  Project Cube
//
//  Created by Tamas Santa on 15/01/15.
//  Copyright (c) 2015 Tamas Santa. All rights reserved.
//

import Foundation

class Ball: Actor {
    var directionX = 1
    var directionY = 1
    
    init(location:Point)
    {
        super.init()
        self.Location = location
        self.Animations.append(LoadAnimation())
        self.CurrentAnimation = self.Animations.first!
    }
    
    func LoadAnimation() -> Animation {
        var serializer = AnimationSerializer()
        var unzippedFile = serializer.ReadZip(NSBundle.mainBundle().pathForResource("ball", ofType: "pmz")!)
        
        return serializer.Deserialize(unzippedFile!)
    }
}